diff -Naur engage.old/src/e_mod_main.c engage/src/e_mod_main.c
--- engage.old/src/e_mod_main.c	2012-10-10 23:56:05.000000000 +0800
+++ engage/src/e_mod_main.c	2012-10-11 12:06:52.000000000 +0800
@@ -674,12 +674,16 @@
 
    Ngi_Item *it = ng->item_active;
 
-   if (ev->buttons == 2)
+   // Remove this. Can be accessed by
+   // (i) Settings -> Extensions,  (ii) Everything, (iii) Right click @ bar
+   // Instead bind middle mouse button to comp-scale, same class windows
+   /*if (ev->buttons == 2)
      {
         ngi_configure_module(ng->cfg);
         return EINA_TRUE;
      }
-   else if (ng->item_active && ev->buttons == 1)
+   else*/
+   if (ng->item_active && ev->buttons == 1)
      {
         it->drag.x = ev->root.x;
         it->drag.y = ev->root.y;
@@ -1670,7 +1674,7 @@
    // will be rolled into e17 core and this won't be needed
    EINA_LIST_FOREACH(e_config->modules, l, em)
      {
-	if (!strcmp(em->name, "comp"))
+	if (!strcmp(em->name, "comp") && em->enabled)
 	  {
 	     have_comp = 1;
 	     break;
diff -Naur engage.old/src/ng_taskbar.c engage/src/ng_taskbar.c
--- engage.old/src/ng_taskbar.c	2012-10-10 23:56:05.000000000 +0800
+++ engage/src/ng_taskbar.c	2012-10-11 12:06:26.000000000 +0800
@@ -633,7 +633,7 @@
 _item_set_label(Ngi_Item_Taskbar *it)
 {
    const char *title;
-   unsigned int max_len = 36;
+   unsigned int max_len = 25;
 
    title = e_border_name_get(it->border);
 
@@ -705,7 +705,8 @@
           } /* switch */
         ngi_border_menu_show(item->box, it->border, x, y, dir, ev->timestamp);
      }
-   else if (ev->buttons == 1 && ev->double_click && !it->border->iconic)
+   else if (ev->buttons == 1 && !ev->double_click && bd->focused &&
+            bd->desk == e_desk_current_get(bd->zone))
      {
         e_border_iconify(it->border);
      }
@@ -732,7 +733,17 @@
 	ngi_bar_lock(item->box->ng, 0);
      }
 
-   if (ev->buttons != 1)
+   if (ev->buttons == 2 && (bd) && (bd->client.icccm.class))
+     {
+        char buf[1024];
+        E_Action *act = e_action_find("scale-windows");
+        if (!act) return;
+
+        snprintf(buf, 1024, "go_scale_class:%s", bd->client.icccm.class);
+        act->func.go(NULL, buf);
+        return;
+     }
+   else if (ev->buttons != 1)
      return;
 
    if (bd->shaded)
@@ -757,19 +768,7 @@
      {
 	e_border_raise(bd);
      }
-   if (bd->focused)
-     {
-	char buf[1024];
-	E_Action *act = e_action_find("scale-windows");
-
-	if (!act) return;
-	if (!bd->client.icccm.class) return;
-	  
-	snprintf(buf, 1024, "go_scale_class:%s", bd->client.icccm.class);
-	act->func.go(NULL, buf);
-     }
-   else
-     e_border_focus_set(bd, 1, 1);
+   e_border_focus_set(bd, 1, 1);
 }
 
 
