--- textext.py.orig	2012-08-22 23:46:54.000000000 +0800
+++ textext.py	2012-05-28 19:42:03.000000000 +0800
@@ -875,7 +875,8 @@
         return PdfConverterBase.convert(self, *a, **kw)
 
     def pdf_to_svg(self):
-        exec_command(['pdf2svg', self.tmp('pdf'), self.tmp('svg'), '1'])
+        #exec_command(['pdf2svg', self.tmp('pdf'), self.tmp('svg'), '1'])
+        exec_command(['pdf2svg', self.tmp('pdf'), self.tmp('svg')])
 
     def get_transform(self, scale_factor):
         return 'scale(%f,%f)' % (scale_factor, scale_factor)
