diff -Naur net/src/e_mod_gadcon.c net.new/src/e_mod_gadcon.c
--- net/src/e_mod_gadcon.c	2011-11-20 02:10:45.372473213 +0800
+++ net.new/src/e_mod_gadcon.c	2012-01-14 15:45:42.866556470 +0800
@@ -81,8 +81,8 @@
 static void 
 _gc_orient(E_Gadcon_Client *gcc, E_Gadcon_Orient orient) 
 {
-   e_gadcon_client_aspect_set(gcc, 16, 16);
-   e_gadcon_client_min_size_set(gcc, 16, 16);
+   e_gadcon_client_aspect_set(gcc, 32, 16);
+   e_gadcon_client_min_size_set(gcc, 32, 16);
 }
 
 static const char *
